controllers/managedocs_controller.go

```diff
@@ -1754,6 +1885,21 @@ func (r *ManagedOCSReconciler) deleteCSVByPrefix(name string) error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) getCSVByPrefix(name string) (*opv1a1.ClusterServiceVersion, error) {
+	csvList := opv1a1.ClusterServiceVersionList{}
+	if err := r.list(&csvList); err != nil {
+		return nil, fmt.Errorf("unable to list csv resources: %v", err)
+	}
+
+	for index := range csvList.Items {
+		candidate := &csvList.Items[index]
+		if strings.HasPrefix(candidate.Name, name) {
+			return candidate, nil
+		}
+	}
+	return nil, fmt.Errorf("unable to find the csv prefixed with %s", name)
+}
```

Comment:
This is an extract from `deleteCSVByPrefix` so please refactor `deleteCSVByPrefix` to use this method internally so we will not need to solve bugs in 2 distinct places.